extensions/sonarcloud/templates/maven/template.json

```diff
@@ -0,0 +1,104 @@
+{
+  "id": "sonarcloud_maven_template",
+  "name": "Maven with SonarCloud",
+  "category": "Build",
+  "iconTaskId": "ce096e50-6155-4de8-8800-4221aaeed4a1",
+  "defaultHostedQueue": null,
+  "description": "Build your Java projects and run tests with Apache Maven, and analyze with SonarCloud. This template requires Maven to be installed on the build agent.",
+  "template": {
+    "build": [
+      {
+        "environment": {},
+        "enabled": true,
+        "continueOnError": false,
+        "alwaysRun": false,
+        "displayName": "Prepare analysis on SonarCloud",
+        "timeoutInMinutes": 0,
+        "task": {
+          "id": "14d9cde6-c1da-4d55-aa01-2965cd301255",
+          "versionSpec": "1.*",
+          "definitionType": "task"
+        },
+        "inputs": {
+          "scannerMode": "Other",
+          "configMode": "file",
+          "configFile": "sonar-project.properties",
+          "cliProjectKey": "",
+          "projectKey": "",
+          "cliProjectName": "",
+          "projectName": "",
+          "cliProjectVersion": "1.0",
+          "projectVersion": "1.0",
+          "cliSources": ".",
+          "extraProperties": "# Additional properties that will be passed to the scanner, \n# Put one key=value per line, example:\n# sonar.exclusions=**/*.bin"
+        }
+      },
+      {
+        "environment": {},
+        "enabled": true,
+        "continueOnError": false,
+        "alwaysRun": false,
+        "timeoutInMinutes": 0,
+        "task": {
+          "id": "ac4ee482-65da-4485-a532-7b085873e532",
+          "versionSpec": "2.*",
+          "definitionType": null
+        },
+        "inputs": {
+          "mavenPOMFile": "$(Parameters.mavenPOMFile)",
+          "goals": "package",
```

Comment:
I wonder if it is good to specify values here. If you don't, will it fallback to use default value? If yes, then in will be better to not fix the value in the template, but instead rely on the task defaults. This way, if they later do some changes to their default, our template will stay "up to date".

Reply:
I've removed _goals_ but left _mavenPOMFile_ as it refers to a process level-variable.